### PR TITLE
feat: support reductions in lazyframe.select for Dask

### PR DESCRIPTION
**Pull request by MarcoGorelli**
🤔 getting closer... ~not totally happy with this though as `df.select(nw.col('a').sum().alias('a_sum'))` wouldn't work~

## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue # 
- Closes #

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added 
- [ ] Documented the changes

### CODE REVIEW

**Comment by FBruzzesi** [comment]
When CI fails, but it's actually a good sign and we are happy 😂

DancingCatKittenGIF

File: narwhals/_dask/namespace.py [hunk 1]
```diff
@@ -97,5 +125,6 @@ def _create_expr_from_callable(  # pragma: no cover
             function_name=function_name,
             root_names=root_names,
             output_names=output_names,
+            is_scalar=False,  # TODO(marco): check!
```
Comment (MarcoGorelli) [inline narwhals/_dask/namespace.py:128]: gonna have to check this one before merging

File: narwhals/_dask/expr.py [hunk 2]
```diff
@@ -69,6 +73,7 @@ def _from_call(
         call: Any,
         expr_name: str,
         *args: Any,
+        is_scalar: bool,
```
Comment (FBruzzesi) [inline narwhals/_dask/expr.py:76]: One tiny and nitty objection on the naming: what about `returns_scalar`? 

- it is already used in the pandas-like API (for a similar reason/behavior)
- Expr itself is _not_ a scalar, but _returns_ one

**Comment by MarcoGorelli** [comment]
Right, let's do this to enable other prs to get through

**Pull request merged**